src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AppConfigurationClient, AppConfigurationClientOptions } from "@azure/app-configuration";
+import { ConfigurationClientWrapper } from "./ConfigurationClientWrapper.js";
+import { TokenCredential } from "@azure/identity";
+import { AzureAppConfigurationOptions, MaxRetries, MaxRetryDelayInMs } from "./AzureAppConfigurationOptions.js";
+import { isFailoverableEnv } from "./requestTracing/utils.js";
+import * as RequestTracing from "./requestTracing/constants.js";
+
+const TCP_ORIGIN_KEY_NAME = "_origin._tcp";
+const ALT_KEY_NAME = "_alt";
+const TCP_KEY_NAME = "_tcp";
+const ENDPOINT_KEY_NAME = "Endpoint";
+const ID_KEY_NAME = "Id";
+const SECRET_KEY_NAME = "Secret";
+const AZCONFIG_DOMAIN_LABEL = ".azconfig.";
+const APPCONFIG_DOMAIN_LABEL = ".appconfig.";
```

Comment:
```suggestion
const TRUSTED_DOMAIN_LABELS = [".azconfig.", ".appconfig."];
```

Then you won't need line 257